### PR TITLE
chore: create wallet-api deploy

### PR DESCRIPTION
**Pull request by bernokl**
# Context
Add deployment for wallet-api service

### CODE REVIEW

File: nix/cardano-services/deployments/wallet-api.nix [hunk 1]
```diff
@@ -0,0 +1,95 @@
+{
+  config,
+  values,
+  lib,
+  utils,
+  chart,
+  ...
+}: {
+  providers.wallet-api-provider = {
```
Comment (iccicci) [inline nix/cardano-services/deployments/wallet-api.nix:9]: We have no `wallet-api-provider` ready yet: please remove this

File: nix/cardano-services/deployments/default.nix [hunk 2]
```diff
@@ -822,10 +834,24 @@ in
...
+          projectors = {
+            wallet-api.enabled = true;
+            handle.enabled = true;
+            stake-pool.enabled = true;
+          };
+
+
           values = {
+            pg-boss-worker.enabled = true;
+            pg-boss-worker.queues = "pool-metadata,pool-metrics";
```
Comment (iccicci) [inline nix/cardano-services/deployments/default.nix:854]: This section seems to be intended to align the env with other env, the only line actually relevant to new projection is `wallet-api.enabled = true;`; if that's true I'm fine, otherwhise I would need some more clarifications.

In any case, the proper value for last line is `pool-delist-schedule,pool-metadata,pool-metrics,pool-rewards`.

File: nix/cardano-services/deployments/wallet-api.nix [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+    };
+
+    args = ["start-projector"];
+    port = 3000;
+
+    env = {
+      NETWORK = config.network;
+      LOGGER_MIN_SEVERITY = values.cardano-services.loggingLevel;
+      OGMIOS_SRV_SERVICE_NAME = values.backend.ogmiosSrvServiceName;
+      PROJECTION_NAMES = "wallet-api";
```
[the remark this answers is not in the document]
Comment (bernokl) [reply nix/cardano-services/deployments/wallet-api.nix:73]: updated, sorry for the misunderstanding

File: nix/cardano-services/deployments/wallet-api.nix (hunk 1 above)
Comment (bernokl) [reply nix/cardano-services/deployments/wallet-api.nix:9]: This was part of my template testing, it has been removed

File: nix/cardano-services/deployments/default.nix (hunk 2 above)
Comment (bernokl) [reply nix/cardano-services/deployments/default.nix]: Fixed thank you

File: nix/cardano-services/deployments/default.nix [hunk 4]
```diff
@@ -96,6 +101,11 @@ in
           resources.requests = mkPodResources "150Mi" "100m";
         };
 
+        wallet-api = {
```
[the remark this answers is not in the document]
Comment (bernokl) [reply nix/cardano-services/deployments/default.nix:99]: Removed

[the remark this answers is not in the document]
Comment (iccicci) [reply nix/cardano-services/deployments/default.nix:99]: Could it be you removed the `wallet-api-projector` section (the one actually required) and you left the `wallet-api` section (which, if I understood correctly is for the `provider-server` will still don't have)?

**Review by iccicci — CHANGES_REQUESTED** [review]

File: nix/cardano-services/deployments/default.nix (hunk 4 above)
[the remark this answers is not in the document]
Comment (bernokl) [reply nix/cardano-services/deployments/default.nix:99]: No it is confirmed correct. Line 88 is where projectors starts so 99 is in the projector section, I removed 75 which was under providers section

[the remark this answers is not in the document]
Comment (iccicci) [reply nix/cardano-services/deployments/default.nix:99]: Thank you both for checking again and for the additional info!
Sorry for the confusion, but I'm still not so strong on nix files.

**Review by iccicci — APPROVED** [review]
Awesome! Thank you very much!

**Review by gytis-ivaskevicius — APPROVED** [review]

**Review by michalrus — COMMENTED** [review]
LGTM (post factum)

**Pull request merged**